'EXIV2API' does not name a type- when building with AUR default flags

#### **Describe the bug**
exiv2 fails to build (either with the Arch Linux's 'exiv2-git' AUR package or by directly cloning this repo with the following error:

```
[  9%] Building CXX object src/CMakeFiles/exiv2lib_int.dir/casiomn_int.cpp.o
In file included from /home/lito/.local/build/exiv2/src/casiomn_int.cpp:9:
/home/lito/.local/build/exiv2/src/i18n.h:36:1: error: 'EXIV2API' does not name a type
   36 | EXIV2API const char* _exvGettext(const char* str);
      | ^~~~~~~~
/home/lito/.local/build/exiv2/src/i18n.h:36:1: note: the macro 'EXIV2API' had not yet been defined
In file included from /home/lito/.local/build/exiv2/include/exiv2/tags.hpp:7,
                 from /home/lito/.local/build/exiv2/src/casiomn_int.cpp:11:
/home/lito/.local/build/exiv2/build/exiv2lib_export.h:12:15: note: it was later defined here
   12 | #      define EXIV2API __attribute__((visibility("default")))
      |               ^~~~~~~~
make[2]: *** [src/CMakeFiles/exiv2lib_int.dir/build.make:93: src/CMakeFiles/exiv2lib_int.dir/casiomn_int.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1317: src/CMakeFiles/exiv2lib_int.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```
 

#### **To Reproduce**
Steps to reproduce the behavior:
1. In Arch, try to compile the exiv2-git package from the AUR.
2. Otherwise, clone this repo and try to build it with these flags:

```
cmake -B build \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DCMAKE_SKIP_RPATH=ON \
    -DEXIV2_BUILD_SAMPLES=ON \
    -DEXIV2_BUILD_UNIT_TESTS=ON \
    -DEXIV2_ENABLE_VIDEO=ON \
    -DEXIV2_ENABLE_NLS=ON \
    -DEXIV2_ENABLE_XMP=ON \
    -DEXIV2_ENABLE_CURL=ON \
    -DEXIV2_ENABLE_WEBREADY=ON
```

#### **Expected behavior**
exiv2 should build successfully both from the package and from this repo with these flags

#### **Desktop (please complete the following information):**
- OS and version: Arch Linux
- Exiv2 version and source: commit [5c960c4](https://github.com/Exiv2/exiv2/commit/5c960c45bd2655f3ce30b4c1741641b6502435a0) from this repo
- Compiler and version: Both GCC 15.1.1 and Clang 20.1.8
- Compilation mode and/or compiler flags: Both with default flags and with compiler optimization flags:
```
-march=znver3 -mtune=znver3 -O3 -pipe -fno-plt -fexceptions \
        -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security \
        -fstack-clash-protection -fcf-protection \
        -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer \
        -fopenmp -pthread
```

#### **Additional context**
What I'm actually trying to do is to compile plasma-desktop-git, which depends on the aforementioned exiv2-git package.

--- a/app/exiv2.cpp
+++ b/app/exiv2.cpp
@@ -35,6 +35,10 @@ namespace fs = std::filesystem;
 #define _strdup strdup
 #endif
 
+#ifdef EXV_ENABLE_NLS
+#include <libintl.h>
+#endif
+
 // *****************************************************************************
 // local declarations
 namespace {

--- a/src/i18n.h
+++ b/src/i18n.h
@@ -30,12 +30,12 @@
 
 /* NLS can be disabled through the configure --disable-nls option. */
 #ifdef EXV_ENABLE_NLS
-#include <libintl.h>
+#include "types.hpp"
 
 // Definition is in types.cpp
-EXIV2API const char* _exvGettext(const char* str);
+const char* _exvGettext(const char* str);
 
-#define _(String) _exvGettext(String)
+#define _(String) Exiv2::exvGettext(String)
 #define N_(String) String
 
 #else /* NLS is disabled */

--- a/src/types.cpp
+++ b/src/types.cpp
@@ -19,6 +19,10 @@
 #include <filesystem>
 namespace fs = std::filesystem;
 
+#ifdef EXV_ENABLE_NLS
+#include <libintl.h>
+#endif
+
 // *****************************************************************************
 namespace {
 //! Information pertaining to the defined %Exiv2 value type identifiers.

